powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java

```diff
@@ -0,0 +1,138 @@
+
+import com.google.common.io.BaseEncoding;
+import com.wultra.security.powerauth.client.v3.*;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.push.client.PushServerTestClientFactory;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.CryptoProviderException;
+import io.getlime.security.powerauth.crypto.lib.util.KeyConvertor;
+import org.springframework.context.annotation.Profile;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.KeyPair;
+import java.util.*;
+
+/**
+ * Mock controller for PowerAuth services.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@RestController
+@RequestMapping("powerauth-java-server/rest/v3")
+@Profile("test")
+public class MockPowerAuthController {
+
+    private final Set<String> blockedActivations = new HashSet<>();
+
+    @PostMapping("application/list")
+    public ObjectResponse<GetApplicationListResponse> getApplicationList() {
+        final GetApplicationListResponse response = new GetApplicationListResponse();
+        final GetApplicationListResponse.Applications app = new GetApplicationListResponse.Applications();
+        app.setApplicationId(PushServerTestClientFactory.TEST_APPLICATION_NAME);
+        response.getApplications().add(app);
+        return new ObjectResponse<>(response);
+    }
+
+    @PostMapping("application/create")
+    public ObjectResponse<CreateApplicationResponse> createApplication() {
+        final CreateApplicationResponse response = new CreateApplicationResponse();
+        return new ObjectResponse<>(response);
+    }
+
+    @PostMapping("application/detail")
+    public ObjectResponse<GetApplicationDetailResponse> getApplicationDetail() {
+        final GetApplicationDetailResponse response = new GetApplicationDetailResponse();
+        response.setApplicationId(PushServerTestClientFactory.TEST_APPLICATION_NAME);
+        final GetApplicationDetailResponse.Versions version = new GetApplicationDetailResponse.Versions();
+        version.setApplicationVersionId(PushServerTestClientFactory.TEST_APPLICATION_VERSION);
+        version.setSupported(true);
+        version.setApplicationKey("key");
+        version.setApplicationSecret("secret");
+        response.getVersions().add(version);
+        try {
+            final KeyPair keyPair = new KeyGenerator().generateKeyPair();
+            final KeyConvertor keyConvertor = new KeyConvertor();
+            response.setMasterPublicKey(BaseEncoding.base64().encode(keyConvertor.convertPublicKeyToBytes(keyPair.getPublic())));
+        } catch (CryptoProviderException e) {
+            // Ignore error in mock endpoint
```

Comment:
Or at least
```suggestion
            logger.error("Crypto provider error", e);
```

Reply:
Yes, logging is fine, but again, the exception cannot happen.